spot-ml/ml_ops.py

```diff
@@ -0,0 +1,203 @@
+    LOCAL_WORDRESULTS = "{0}/word_results.csv".format(LPATH)
+
+    HDFS_SCORED_CONNECTS = "{0}/scores".format(HPATH)
+    HDFS_MODEL = "{0}/model".format(HPATH)
+
+    LDA_OUTPUT_DIR = "{1}/{1}".format(args.type,args.fdate)
+
+    #get nodes and create comma seperated list
+    NODES = conf.get('DEFAULT','NODES').split()
+    nodes_csl = ','.join(NODES)
+
+    cmd = "hdfs dfs -rm -R -f {0}".format(HDFS_WORDCOUNTS)
+    execute_cmd(cmd,logger)
+
+    cmd = "mkdir -p {0}".format(LPATH)
+    execute_cmd(cmd,logger)
+
+    # protect the flow_scores.csv file
+    cmd = "rm -f {0}/*.{dat,beta,gamma,other,pkl}".format(LPATH)
+    execute_cmd(cmd,logger)
+
+    cmd = "hdfs dfs -rm -R -f {0}".format(HDFS_SCORED_CONNECTS)
+    execute_cmd(cmd,logger)
+
+    # Add -p <command> to execute pre MPI command.
+    # Pre MPI command can be configured in /etc/spot.conf
+    # In this script, after the line after --mpicmd ${MPI_CMD} add:
+    # --mpiprep ${MPI_PREP_CMD}
+
+    if conf.get('mpi',"MPI_PREP_CMD"):
+        cmd = conf.get('mpi',"MPI_PREP_CMD")
+        execute_cmd(cmd,logger)
+
+    SPK_CONFIG = conf.get('spark','SPK_CONFIG')
+    SPK_DRIVER_MEM = conf.get('spark','SPK_DRIVER_MEM')
+    SPK_EXEC = conf.get('spark','SPK_EXEC')
+    SPK_EXEC_CORES = conf.get('spark','SPK_EXEC_CORES')
+    SPK_EXEC_MEM = conf.get('spark','SPK_EXEC_MEM')
+    SPK_EXEC_MEM_OVERHEAD = conf.get('spark', 'SPK_EXEC_MEM_OVERHEAD')
+    SPK_DRIVER_MAX_RESULTS = conf.get('spark', 'SPK_DRIVER_MAX_RESULTS')
+    SPK_DRIVER_MEM_OVERHEAD = conf.get('spark', 'SPK_DRIVER_MEM_OVERHEAD')
+    LDAPATH = conf.get('DEFAULT','LDAPATH')
+    LUSER = conf.get('DEFAULT','LUSER')
+    MPI_CMD = conf.get('mpi','MPI_CMD')
+    PROCESS_COUNT = conf.get('mpi','PROCESS_COUNT')
+    TOPIC_COUNT = conf.get('DEFAULT','TOPIC_COUNT')
+
+    if tol:
+        TOL = tol
+    else:
+        TOL = conf.get('DEFAULT','TOL')
+
+    #prepare options for spark-submit
+    spark_cmd = [
+        "time", "spark-submit", "--class org.apache.spot.SuspiciousConnects",
+         "--master yarn-client", "--conf spark.driver.maxPermSize=512m", "--conf spark.driver.cores=1",
+         "--conf spark.dynamicAllocation.enabled=true", "--conf spark.dynamicAllocation.minExecutors=1",
+         "--conf spark.executor.extraJavaOptions=-XX:MaxPermSize=512M -XX:PermSize=512M",
+         "--conf spark.shuffle.io.preferDirectBufs=false", "--conf spark.kryoserializer.buffer.max=512m",
+         "--conf spark.shuffle.service.enabled=true", "--conf spark.yarn.am.waitTime=1000000"]
+
+    spark_extras = [
+        "--driver-memory " + SPK_DRIVER_MEM,
+        "--conf spark.dynamicAllocation.maxExecutors=" + SPK_EXEC,
+        "--conf spark.executor.cores=" + SPK_EXEC_CORES,
+        "--conf spark.executor.memory=" + SPK_EXEC_MEM,
+        "--conf spark.driver.maxResultSize=" + SPK_DRIVER_MAX_RESULTS,
+        "--conf spark.yarn.driver.memoryOverhead=" + SPK_DRIVER_MEM_OVERHEAD,
```

Comment:
`spark.yarn.driver.memoryOverhead` should be `spark.yarn.am.memoryOverhead` based on current spot branch.